File: src/main/java/net/explorviz/trace/persistence/PersistenceSpan.java
package net.explorviz.trace.persistence;

public record PersistenceSpan(
    String landscapeToken,
    String gitCommitChecksum,
    String spanId,
    String parentSpanId,
    String traceId,
    long startTime,
    long endTime,
    String nodeIpAddress, // TODO: Convert into InetAddress type?
    String hostName,
    String applicationName,
    String applicationLanguage,
    String applicationInstance,
    String filePath,
    String functionName,
    String className,
    String k8sPodName,
    String k8sNodeName,
    String k8sNamespace,
    String k8sDeploymentName
) {
  public static final String DEFAULT_LANGUAGE = "LANGUAGE_UNSPECIFIED";
}

File: src/main/java/net/explorviz/trace/persistence/PersistenceSpan.kt
package net.explorviz.trace.persistence

data class PersistenceSpan(
    val landscapeTokenId: String,
    val landscapeTokenSecret: String,

    val traceId: String?,
    val spanId: String?,
    val parentSpanId: String?,

    val startTime: Long,
    val endTime: Long,

    val applicationName: String,

    val spanEntity: SpanEntity,
)
